### DISCUSSION TITLE
Tests are failing on bazel.

### DISCUSSION THREAD

**Discussion by Vishwajith-Shettigar** [oppia/oppia-android#5371] [Q&A (Other)] [answered]
I'm facing difficulty running tests on Bazel; all the test suites are failing. I've attached the test log. Can anybody help with it please?

Bazel version 4.0.0 and java 8.

[test.log](https://github.com/oppia/oppia-android/files/14767006/test.log)

**Comment by BenHenning** [comment, marked as answer]
@Vishwajith-Shettigar can you provide some more context on your environment? Specifically:
- Can you confirm you followed all of the Bazel setup instructions for your OS platform (per the wiki)?
- Can you confirm which branch & commit hash you have checked out & whether it includes extra changes?
- Can you confirm which OS platform you're using?
- Can you share the results of running ``bazel info`` in your local repository via a terminal?
- Can you share the results of running ``java -version`` in the same directory?

These are details that I think would add significant clarity to your question and hopefully provide enough context to answer it.

That being said, I suspect there's either a SecurityManager situation happening here or there are some incompatibilities in ``Unsafe`` (which might come from incompatible JDK versions/vendors). It's hard to narrow this down without the details requested above.

**Reply by Vishwajith-Shettigar** [reply]
- Yes,i followed all wiki instructions.
- I tried running test from develop branch with no changes made
- Windows, and for bazel Im using WSL.
- ![image](https://github.com/oppia/oppia-android/assets/76042077/663d639e-51ea-450a-b1f7-2805f5d0108c)
- 
![image](https://github.com/oppia/oppia-android/assets/76042077/059e71fd-e94b-45e8-9833-5d03cacd6464)

**Reply by Vishwajith-Shettigar** [reply]
Some tests are passing, i tried running utility/src/test/java/org/oppia/android/util/math tests, 13 failed and 6 passed  -- FYI

![Screenshot 2024-03-28 200821](https://github.com/oppia/oppia-android/assets/76042077/8c106091-1c18-4a59-aecc-44630a2fce95)
